aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2520,6 +2521,147 @@ Tensor & transpose_(Tensor & self, int64_t dim0, int64_t dim1) {
   return self;
 }
 
+namespace {
+// Transpose implementation for sparse compressed layouts
+// NB: We assume that dim1,dim0 have already been wrapped
+static inline Tensor sparse_compressed_transpose(
+    const Tensor& self,
+    int64_t dim0,
+    int64_t dim1) {
+  auto compressed_inds = AT_DISPATCH_ROW_SPARSE_COMPRESSED_LAYOUTS(
+      self.layout(),
+      "compressed_inds",
+      [&self]() { return self.crow_indices(); },
+      [&self]() { return self.ccol_indices(); });
```

Comment:
I have noticed a pattern of cloning indices for result tensors. These accessors do that, don't they?

Reply:
I have noticed that pattern as well. 

It looks like `self.c<col/row>_indices` will resolve to `get_sparse_csr_impl(self)->compressed_indices().alias()`. So not a clone, but I don't think we *need* to clone indices here as the result is expected to be a view.